torch_xla/csrc/tensor.cpp

```diff
@@ -725,7 +725,11 @@ void XLATensor::SetInPlaceIrValue(torch::lazy::Value ir_value) {
 void XLATensor::AssignIrValue(torch::lazy::Value ir_value) const {
   // Sharding annotation is not null, if xla_data() is sharded.
   ShardingSpecPtr sharding = sharding_spec();
-  if (ir_value && sharding != nullptr) {
+  if (CurrentXlaData() != nullptr && sharding != nullptr) {
+    if (!ir_value) {
+      ir_value = CreateTensorNode(CurrentXlaData(), /*read_only=*/false);
+    }
+    XLA_CHECK(ir_value.node != nullptr) << "Tyring to access a null cursor";
```

Comment:
hmm, so this line is to force the `sharding` on the old IR before it is being replaced? I am confuse because we did not clear the sharding after the new `ir_value` being assigned.

Reply:
The intent is to force the same `sharding` on the new IR (input), per a common request like this, `AssignIrValue(torch::lazy::Value())`. Did I get it reversed?

Reply:
oh ok, I read it backward, you re trying to populate the sharding spec to the new IR being assigned.